Index: lib/python/pydispatch/dispatcher.py
===================================================================
--- lib/python/pydispatch/dispatcher.py	(revision 73073)
+++ lib/python/pydispatch/dispatcher.py	(working copy)
@@ -388,7 +388,7 @@
     else:
         for senderkey in backSet:
             try:
-                signals = connections[senderkey].keys()
+                signals = list(connections[senderkey])
             except KeyError:
                 pass
             else:
Index: lib/python/pydispatch/robustapply.py
===================================================================
--- lib/python/pydispatch/robustapply.py	(revision 73073)
+++ lib/python/pydispatch/robustapply.py	(working copy)
@@ -54,7 +54,7 @@
     if not (codeObject.co_flags & 8):
         # fc does not have a **kwds type parameter, therefore
         # remove unacceptable arguments.
-        for arg in named.keys():
+        for arg in list(named):
             if arg not in acceptable:
                 del named[arg]
     return receiver(*arguments, **named)
